src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax;
+
+public class Main {
+    private static FigureSupplier figureSupplier = new FigureSupplier();
```

Comment:
there is no need to declare this static variable

since you're using FigureSupplier in the main method - you can create its instance directly there

making such variables makes sense when you use them in methods, that are often invoked (which is not the case of main method) - this way you avoid creation of new instances each time the method is invoked